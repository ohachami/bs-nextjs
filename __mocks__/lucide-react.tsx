// Mock all icons with a simple mock component
export const ChartLine = () => <div data-testid="mock-chart-line" />;
export const ClipboardList = () => <div data-testid="mock-clipboard-list" />;
export const GitCompareArrows = () => (
  <div data-testid="mock-git-compare-arrows" />
);
export const MoveRight = () => <div data-testid="mock-move-right" />;
