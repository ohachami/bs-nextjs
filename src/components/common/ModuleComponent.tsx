import clsx from 'clsx';
import { MoveRight } from 'lucide-react';
import { ReactNode } from 'react';
import Link from 'next/link';

type Props = {
  title: string;
  description: string;
  variant: 'primary' | 'secondary' | 'tertiary';
  icon: React.ComponentType<React.SVGProps<SVGSVGElement>>;
  to?: string;
  children?: ReactNode;
};

const ModuleComponent: React.FC<Props> = ({
  variant,
  icon: Icon,
  title,
  to,
  description,
  children,
}) => {
    const Content = () => (
        <div className="my-[80px] p-4 bg-white rounded-lg shadow-sm flex flex-col h-full max-h-[387px]">
            <div
                className={clsx(
                    '-mt-[90px] h-[150px] w-[150px] border-[1.5px] rounded-[20px] p-5',
                    {
                        'bg-primary-light border-primary/40': variant === 'primary',
                        'bg-secondary-light border-secondary/40': variant === 'secondary',
                        'bg-tertiary-light border-tertiary/40': variant === 'tertiary',
                    }
                )}
            >
                <Icon
                    className={clsx({
                        'w-full h-full': true,
                        'text-primary': variant === 'primary',
                        'text-secondary': variant === 'secondary',
                        'text-tertiary': variant === 'tertiary',
                    })}
                    strokeWidth={1}
                />
            </div>

            <div className="grow flex flex-col justify-between gap-6 p-6">
                <h1 className="font-geist font-semibold text-[28px] text-card-foreground leading-7 tracking-[-0.6px]">
                    {title}
                </h1>

                <p className="font-geist font-normal text-xs leading-4 text-muted-foreground">
                    {description}
                </p>

                {children && children}

                <div className="w-9 h-9">
                    {to && (
                        <MoveRight
                            strokeWidth={1}
                            size={36}
                            className={clsx({
                                'text-primary': variant === 'primary',
                                'text-secondary': variant === 'secondary',
                                'text-tertiary': variant === 'tertiary',
                            })}
                        />
                    )}
                </div>
            </div>
        </div>
    )

    return to ? <Link href={to}><Content/></Link> : <Content/>;
};

export default ModuleComponent;
