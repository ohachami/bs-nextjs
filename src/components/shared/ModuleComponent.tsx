import { ChartLine } from 'lucide-react'
import { Button } from "@/components/ui/button"
import { MoveRight } from 'lucide-react'

type Props = {
    title: string;
    description: string;
}

const ModuleComponent: React.FC<Props> = ({title, description}) => (
    <div className="p-4 bg-white rounded-lg shadow-sm mt-[76.2px] h-[332px] flex flex-col">
        <div className="h-[152.4px] w-[151px] border-[1.5px] rounded-[20px] border-primary-light bg-[#E2FDE4] p-5 -mt-24">
            <ChartLine className='w-full h-full text-primary' strokeWidth={1} />
        </div>

        <div className='grow flex flex-col justify-between gap-6 p-6'>
            <h1 className='font-geist font-semibold text-[28px] text-card-foreground font-family: Geist; tracking-[-0.6px]'>{title}</h1>

            <p className='font-geist font-normal text-xs leading-4 text-muted-foreground'>
                {description}
            </p>
            
            <Button variant="ghost" className='hover:bg-transparent w-fit p-0'>
                <MoveRight className='!w-9 !h-9 text-[#57D762]' strokeWidth={1} />
            </Button>
        </div>
    </div>
)

export default ModuleComponent