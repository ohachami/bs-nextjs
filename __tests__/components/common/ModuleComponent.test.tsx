import ModuleComponent from '@/components/common/ModuleComponent';
import { render, screen } from '@testing-library/react';
import { ChartLine } from 'lucide-react';

describe('ModuleComponent', () => {
  it('renders title and description', () => {
    const { container } = render(
      <ModuleComponent
        icon={ChartLine}
        title="Market"
        description="A description"
        variant="primary"
      />
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText('Market')).toBeInTheDocument();
    expect(screen.getByText('A description')).toBeInTheDocument();
  });
});

describe('ModuleComponent', () => {
  it('should render a link when `to` prop is provided', () => {
    const { container } = render(
      <ModuleComponent
        icon={ChartLine}
        title="Market"
        description="A description"
        variant="primary"
        to="/modules/tacticalPlanning/exercises"
      />
    );

    const link = container.querySelector('a');
    expect(link).toHaveAttribute('href', '/modules/tacticalPlanning/exercises');
  });

  it('should not render a link when `to` prop is not provided', () => {
    const { container } = render(
      <ModuleComponent
        icon={ChartLine}
        title="Market"
        description="A description"
        variant="primary"
      />
    );

    const link = container.querySelector('a');
    expect(link).toBeNull();
  });
});

describe('ModuleComponent', () => {
  it('should render children when passed', () => {
    render(
      <ModuleComponent
        icon={ChartLine}
        title="Market"
        description="A description"
        variant="primary"
      >
        <p>Additional content</p>
      </ModuleComponent>
    );

    expect(screen.getByText('Additional content')).toBeInTheDocument();
  });

  it('should not render children if not passed', () => {
    render(
      <ModuleComponent
        icon={ChartLine}
        title="Market"
        description="A description"
        variant="primary"
      />
    );

    const children = screen.queryByText('Additional content');
    expect(children).toBeNull();
  });
});
