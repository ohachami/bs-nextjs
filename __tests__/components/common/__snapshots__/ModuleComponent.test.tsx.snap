// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleComponent renders title and description 1`] = `
<div>
  <div
    class="my-[80px] p-4 bg-white rounded-lg shadow-sm flex flex-col h-full max-h-[387px]"
  >
    <div
      class="-mt-[90px] h-[150px] w-[150px] border-[1.5px] rounded-[20px] p-5 bg-primary-light border-primary/40"
    >
      <div
        data-testid="mock-chart-line"
      />
    </div>
    <div
      class="grow flex flex-col justify-between gap-6 p-6"
    >
      <h1
        class="font-geist font-semibold text-[28px] text-card-foreground leading-7 tracking-[-0.6px]"
      >
        Market
      </h1>
      <p
        class="font-geist font-normal text-xs leading-4 text-muted-foreground"
      >
        A description
      </p>
      <div
        class="w-9 h-9"
      />
    </div>
  </div>
</div>
`;
